src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs

```diff
@@ -202,6 +202,14 @@ public IWebHost Build()
                 host.Dispose();
                 throw;
             }
+
+            IServiceProvider GetProviderFromFactory(IServiceCollection collection)
+            {
+                var provider = collection.BuildServiceProvider();
```

Comment:
Hmm after thinking bit, we might need to dispose the temporary container we created If there's an `IServiceProviderFactory<IServiceCollection>` so this code gets a bit more complicated:

```C#
if (factory != null)
{
     using (provider)
     {
          return factory.CreateServiceProvider(factory.CreateBuilder(collection))
     }
}
return provider;
```

Reply:
You don't want to do that. In case factory is resolved from a type and had any disposable dependencies this code will discard these but it might still be used in the provider.
The provider should die quietly as any other Gen0 variable during next partial collection.

Reply:
Any service provider we create should be disposed so that services get disposed. The provider may not be collected if there’s a singleton relying on dispose to unroot itself (we had an example recently where a timer was causing problems). The reasons it’s ok here is because we’re creating a whole new container anyways so disposing services after creation should be fine.

Reply:
I understand